File: backend/booking-api/src/main/java/com/lhsystems/booking/api/dto/FlightDTO.java
package com.lhsystems.booking.api.dto;

import java.time.LocalDateTime;

public record FlightDTO(
        String id,
        String flightNumber,
        String marketingAirline,
        String departureAirport,
        String arrivalAirport,
        LocalDateTime departureTime,
        LocalDateTime arrivalTime,
        Integer travelTimeMinutes,
        Integer stops,
        Integer availableSeats,
        PriceDTO price,
        String status) {}

File: frontend/src/api/models/flightDTO.ts
import type { PriceDTO } from "./priceDTO";
import type { FlightDTOStatus } from "./flightDTOStatus";

export interface FlightDTO {
  /** Unique flight identifier (e.g., LH1234_20260321) */
  id?: string;
  flightNumber?: string;
  marketingAirline?: string;
  departureAirport?: string;
  arrivalAirport?: string;
  departureTime?: string;
  arrivalTime?: string;
  travelTimeMinutes?: number;
  stops?: number;
  stopoverAirport?: string;
  availableSeats?: number;
  price?: PriceDTO;
  status?: FlightDTOStatus;
}
